python/cudf/cudf/_lib/pylibcudf/quantiles.pyx

```diff
@@ -0,0 +1,152 @@
+
+    Parameters
+    ----------
+    input: Column
+        The Column to calculate quantiles on.
+    q: array-like that implements buffer-protocol
+        The quantiles to calculate in range [0,1]
+    interp: Interpolation, default Interpolation.LINEAR
+        The strategy used to select between values adjacent to a specified quantile.
+    ordered_indices: Column, default empty column
+        The column containing the sorted order of input.
+
+        If empty, all input values are used in existing order.
+        Indices must be in range [0, input.size()), but are not required to be unique.
+        Values not indexed by this column will be ignored.
+    exact: bool, default True
+        Returns doubles if True. Otherwise, returns same type as input
+
+    For details, see :cpp:func:`quantile`.
+
+    Returns
+    -------
+    Column
+        A Column containing specified quantiles, with nulls for indeterminable values
+    """
+    cdef:
+        unique_ptr[column] c_result
+        column_view ordered_indices_view
+
+    if ordered_indices is None:
+        ordered_indices_view = column_view()
+    else:
+        ordered_indices_view = ordered_indices.view()
+
+    with nogil:
+        c_result = move(
+            cpp_quantile(
+                input.view(),
+                q,
+                interp,
+                ordered_indices_view,
+                exact,
+            )
+        )
+
+    return Column.from_libcudf(move(c_result))
+
+
+cpdef Table quantiles(
+    Table input,
+    vector[double] q,
+    interpolation interp = interpolation.NEAREST,
+    sorted is_input_sorted = sorted.NO,
+    list column_order = None,
+    list null_precedence = None,
+):
+    """Computes row quantiles with interpolation.
+
+    Computes the specified quantiles by retrieving the row corresponding to the
+    specified quantiles. In the event a quantile lies in between rows, the specified
+    interpolation strategy is used to pick between the rows.
+
+    Parameters
+    ----------
+    input: Table
+        The Table to calculate row quantiles on.
+    q: array-like
+        The quantiles to calculate in range [0,1]
+    interp: Interpolation, default Interpolation.NEAREST
+        The strategy used to select between values adjacent to a specified quantile.
+
+        Must be a non-arithmetic interpolation strategy
+        (i.e. one of
+        {`Interpolation.HIGHER`, `Interpolation.LOWER`, `Interpolation.NEAREST`})
+    is_input_sorted: Sorted, default Sorted.NO
+        Whether the input table has been pre-sorted or not.
+    column_order: list, default None
+        A list of :py:class:`~cudf._lib.pylibcudf.types.Order` enums,
+        indicating the desired sort order for each column.
+        By default, will sort all columns so that they are in ascending order.
+
+        Ignored if `is_input_sorted` is `Sorted.YES`
+    null_precedence: list, default None
+        A list of :py:class:`~cudf._lib.pylibcudf.types.NullOrder` enums,
```

Comment:
Perhaps:

```suggestion
        A list of :py:class:`~.types.NullOrder` enums,
```

Reply:
thanks, updated.

Reply:
nevermind, I don't think sphinx is able to find the enum as a class generally - I'm getting errors where sphinx isn't able to find the reference.
(maybe since the enum doesn't have a docstring).

I changed it back to regular backtics.

Reply:
We will definitely have to do some work on improving documentation of pylibcudf before the end. I'm hoping to find time between now and then to improve enum support in Cython in some of the edge cases that tend to bite us.